source/dofs/dof_tools_sparsity.cc

```diff
@@ -738,19 +777,23 @@ namespace DoFTools
           for (unsigned int f=0; f<GeometryInfo<DoFHandlerType::dimension>::faces_per_cell; ++f)
             support_on_face(i,f) = fe.has_support_on_face(i,f);
 
+        Table<2,bool> int_dof_mask = bool_dof_couplings_from_component_couplings(fe, int_mask);
+
         typename DoFHandlerType::active_cell_iterator cell = dof.begin_active (),
                                                       endc = dof.end ();
         for (; cell!=endc; ++cell)
-          if (cell->is_locally_owned ())
+          if (((subdomain_id == numbers::invalid_subdomain_id)
+               ||
+               (subdomain_id == cell->subdomain_id()))
+              &&
+              cell->is_locally_owned())
             {
               cell->get_dof_indices (dofs_on_this_cell);
               // make sparsity pattern for this cell
-              for (unsigned int i=0; i<fe.dofs_per_cell; ++i)
-                for (unsigned int j=0; j<fe.dofs_per_cell; ++j)
-                  if (int_dof_mask (i,j) != none)
```

Comment:
ah, I see that this is where the comparison is from. Why not build the regular mask and just convert it by hand here?